scripts/openmic_split.py

```diff
@@ -116,26 +178,46 @@ def make_partitions(metadata, labels, seed, num_splits, ratio):
 
     ratio : float in [0, 1]
         Fraction of data to separate for training
+
+    prob_ratio : float in [0, 1]
+        Minimum probability ratio for P(Y | train) (or P(Y | test)) to P(Y)
     '''
-    sample_keys, label_matrix = load_label_matrix(metadata, labels)
 
-    splitter = StratifiedShuffleSplit(n_splits=num_splits,
+    sample_keys, artist_labels, label_matrix = load_label_matrix(metadata, labels, dupe_file)
+
+    splitter = StratifiedShuffleSplit(n_splits=num_splits * 1000,
                                       random_state=seed,
                                       test_size=1-ratio)
 
     # Convert sparse multi-labels to multiclass label array
-    labels = label_matrix.idxmax(axis=1)
+    labels = artist_labels.idxmax(axis=1)
 
     # Loop over folds; we can use the label vector as if it was features here
     # since we never actually look at the "features"
-    for fold, (train_idx, test_idx) in tqdm(enumerate(splitter.split(labels, labels))):
-        train_ser = sample_keys[train_idx].sort_values()
-        test_ser = sample_keys[test_idx].sort_values()
-        if set(train_ser) & set(test_ser):
+    fold = 0
+    for artist_train_idx, artist_test_idx in tqdm(splitter.split(labels, labels)):
+        train_artists = artist_labels.index[artist_train_idx]
+        test_artists = artist_labels.index[artist_test_idx]
+
+        train_idx = sample_keys[sample_keys['artist_id'].isin(train_artists)]['sample_key'].sort_values()
+        test_idx = sample_keys[sample_keys['artist_id'].isin(test_artists)]['sample_key'].sort_values()
+
+        if set(train_idx) & set(test_idx):
```

Comment:
is this taking the intersection? not familiar with this syntax (could have seen this as the union?)

Reply:
Yup, `set(A) & set(B)` = `{x | x in A AND x in B}`.